internal/wrphandlers/qos/priority_queue.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: 2024 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package qos
+
+import (
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+// Item contains a wrp Message pointer and is managed by PriorityQueue.
+type Item struct {
+	value    *wrp.Message
+	priority wrp.QOSValue
+	index    int
+}
+
+// PriorityQueue implements heap.Interface and holds Items, using wrp.QOSValue as its priority.
+// https://xmidt.io/docs/wrp/basics/#qos-description-qos
+type PriorityQueue []*Item
```

Comment:
Why are the Items pointers vs values?  Values are easier to work with.

Reply:
for sure, we can use values